be/src/exec/vectorized/cross_join_node.cpp

```diff
@@ -509,6 +518,91 @@ Status CrossJoinNode::_build(RuntimeState* state) {
+    std::vector<bool> res(ctxs.size(), false);
+    for (int i = 0; i < res.size(); ++i) {
+        auto* conjunct = ctxs[i];
+        // now we can only build runtime filter for two child
+        if (conjunct->root()->children().size() != 2) {
+            continue;
+        }
+        auto left_child = conjunct->root()->get_child(0);
+        auto right_child = conjunct->root()->get_child(1);
+        if (left_child->is_slotref()) {
+            std::vector<SlotId> slots;
+            right_child->get_slot_ids(&slots);
+            // slots check
+            if (build_slots.find(left_child->get_column_ref()->slot_id()) != build_slots.end()) {
+                // left child is in build chunk
+                continue;
+            }
+            for (auto slotId : slots) {
+                if (build_slots.find(slotId) == build_slots.end()) {
+                    continue;
+                }
+            }
+            res[i] = true;
+        }
+    }
+    return res;
+}
+
+StatusOr<std::list<ExprContext*>> CrossJoinNode::rewrite_runtime_filter(ObjectPool* pool,
+                                                                        const std::vector<bool>& could_rewrite,
+                                                                        Chunk* chunk,
+                                                                        const std::vector<ExprContext*>& ctxs) {
+    std::list<ExprContext*> filters;
+    for (int i = 0; i < ctxs.size(); ++i) {
+        if (could_rewrite[i]) {
+            auto* conjunct = ctxs[i];
+            auto left_child = conjunct->root()->get_child(0);
+            auto right_child = conjunct->root()->get_child(1);
+            // all of the child(1) in expr is in build chunk
+            ASSIGN_OR_RETURN(auto res, conjunct->evaluate(right_child, chunk));
+            DCHECK_EQ(res->size(), 1);
+            ColumnPtr col;
+            if (res->is_constant()) {
+                col = res;
+            } else if (res->is_nullable()) {
+                if (res->is_null(0)) {
+                    col = ColumnHelper::create_const_null_column(1);
+                } else {
+                    auto data_col = down_cast<NullableColumn*>(res.get())->data_column();
+                    col = std::make_shared<ConstColumn>(data_col, 1);
+                }
+            } else {
+                col = std::make_shared<ConstColumn>(res, 1);
+            }
+
+            auto literal = pool->add(new VectorizedLiteral(std::move(col), right_child->type()));
+            auto new_expr = conjunct->root()->clone(pool);
+            auto new_left = left_child->clone(pool);
+            new_expr->clear_children();
+            new_expr->add_child(new_left);
+            new_expr->add_child(literal);
+            filters.push_back(new ExprContext(new_expr));
+        }
+    }
+    return filters;
+}
+
+Status CrossJoinNode::_build_runtime_filter(RuntimeState* state) {
+    if (_build_chunk && _build_chunk->num_rows() == 1) {
```

Comment:
can this optimization be apply to the build chunk that has more rows, for example 2 rows, 3 rows.

Reply:
we could did it later.
if right table has two rows and rows is [1, 2]
predicate could rewrite as A > 1 or A > 2